lib/smart_proxy_pulp_plugin/pulp_api.rb

```diff
@@ -17,5 +17,12 @@ class Api < Sinatra::Base
         log_halt 503, "Pulp server '#{URI.parse(::PulpProxy::Plugin.settings.pulp_url.to_s).host}' is unknown"
       end
     end
+
+    get "/status/diskfree" do
+      disk_raw_data = `df -h #{PulpProxy::Plugin.settings.pulp_dir}`.split("\n").last
```

Comment:
this should not be in the API classes, abstract it to a class (that can also be added tests for)